templates/smtp.rb

```diff
@@ -1,9 +1,20 @@
+if ENV['SMTP_PROVIDER'] == 'mandrill'
+  SMTP_USERNAME = ENV.fetch 'MANDRILL_USERNAME'
+  SMTP_PASSWORD = ENV.fetch 'MANDRILL_APIKEY'
+elsif ENV['SMTP_PROVIDER'] == 'sendgrid'
+  SMTP_USERNAME = ENV.fetch 'SENDGRID_USERNAME'
+  SMTP_PASSWORD = ENV.fetch 'SENDGRID_PASSWORD'
+else
+  SMTP_USERNAME = ENV.fetch 'SMTP_USERNAME'
+  SMTP_PASSWORD = ENV.fetch 'SMTP_PASSWORD'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.